src/ucp/core/ucp_worker.c

```diff
@@ -249,7 +249,13 @@ static void ucp_ep_err_pending_purge(uct_pending_req_t *self, void *arg)
     ucp_request_t *req = ucs_container_of(self, ucp_request_t, send.uct);
     ucs_status_t status = (uintptr_t)arg;
 
-    ucp_request_complete_send(req, status);
+    if (req->send.uct_comp.func) {
+        if (--req->send.uct_comp.count == 0) {
```

Comment:
so the fix should be in uct - in case of error, all "completion" objects should be completed, with error status. we cannot assume in UCP that we will find all of them on the pending queue. what will happen in case of single fragment zero-copy/get request which is not on the pending queue?